src/components/Login.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useContext, useState } from "react";
+
+import ErrorMessage from "./ErrorMessage";
+import { UserContext } from "../context/UserContext";
+import { ApiClient } from "../api-client";
+
+const Login = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [, setToken] = useContext(UserContext);
+
+  const submitLogin = async () => {
+    const response = await ApiClient.submitLogin({
+      email: email,
+      password: password,
+    });
+    const data = await response.json();
+
+    if (!response.ok) {
+      setErrorMessage(data.detail);
+    } else {
+      setToken(data.access_token);
+    }
+  };
```

Comment:
I think all that response/error checking logic should live in client, so here you just do

```
const client = ApiClient()
const token = client.submitLogin()
```